AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Support/Helpers/UserPoolSignInHelper.swift

```diff
@@ -54,12 +63,18 @@ struct UserPoolSignInHelper: DefaultLogger {
         case .smsMfa:
             let delivery = challenge.codeDeliveryDetails
             return .init(nextStep: .confirmSignInWithSMSMFACode(delivery, challenge.parameters))
+        case .totpMFA:
+            return .init(nextStep: .confirmSignInWithTOTPCode)
         case .customChallenge:
             return .init(nextStep: .confirmSignInWithCustomChallenge(challenge.parameters))
         case .newPasswordRequired:
             return .init(nextStep: .confirmSignInWithNewPassword(challenge.parameters))
-        case .unknown:
-            throw AuthError.unknown("Challenge not supported", nil)
+        case .selectMFAType:
+            return .init(nextStep: .continueSignInWithMFASelection(challenge.getAllowedMFATypesForSelection))
+        case .setUpMFA:
+            throw AuthError.unknown("Invalid state flow. setUpMFA is handled internally in `SignInState.resolvingTOTPSetup` state.")
```

Comment:
Should we use `AuthError.invalidState` here?

Reply:
Nice catch.. Fixed.